lib/utils.php

```diff
@@ -137,3 +137,23 @@ function gutenberg_experimental_to_kebab_case( $string ) {
 	return strtolower( implode( '-', $matches[0] ) );
 	//phpcs:enable WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
 }
+
+
+/**
+ * Returns true if the request is a non-legacy REST API request.
+ *
+ * Legacy REST requests should still run some extra code for backwards compatibility.
+ *
+ * @todo: replace this function once core WP function is available: https://core.trac.wordpress.org/ticket/42061.
+ *
+ * @return bool
+ */
+function is_rest_api_request() {
+	if ( empty( $_SERVER['REQUEST_URI'] ) ) {
+		return false;
+	}
+
+	$rest_prefix         = trailingslashit( rest_get_url_prefix() );
+	$is_rest_api_request = ( false !== strpos( $_SERVER['REQUEST_URI'], $rest_prefix ) );
+	return $is_rest_api_request;
```

Comment:
Could this just be `return \defined( 'REST_REQUEST' ) && \REST_REQUEST === true`? Or is this code running too early for that to suffice?

Reply:
Exactly. In some cases it's not defined yet. That's how I started 😄

Reply:
In what circumstances is it too early?

Reply:
I'm not sure I can answer this with confidence.

>While these changes make tt1-blocks to show no warnings at all, the warnings seem to differ per theme. Maybe some filter triggers the code?? 🤔 

I hope someone has some thoughts on this and it might be related as I've testing with multiple themes and the logs weren't consistent and didn't pinpoint the reason for that. So it might not be needed if I find out the reason for these different number of triggered logs 🤔

Reply:
> Exactly. In some cases it's not defined yet.

This definitely feels like a code smell. When exactly did you experience this, or with which themes?